### ISSUE TITLE
DatePicker breaks when using one-digit year

### ISSUE THREAD

**Issue by gunnarschmid** [vaadin/flow-components#6771]
### Description

We use DatePicker also for partial dates, e. g. "end of business year" where only day and month are relevant. We initialize the LocalDate object backing the DatePicker with year = 0, but this breaks the component. The set date format is not applied anymore and no other value can be selected. After some testing it turns out that this happens when the year is set to any one-digit number, but it works if the year is >= 10.

![image](https://github.com/user-attachments/assets/ee2beeac-959e-43ca-8f8e-b7870a21f291)

### Expected outcome

The component is expected to work also with years < 10. It should look like this:

![image](https://github.com/user-attachments/assets/5936a4ee-4ecd-4e38-8240-a13a1b5488c1)

### Minimal reproducible example

```
DatePicker dp = new DatePicker("Datum");
dp.setValue(LocalDate.of(0, 3, 1));
dp.setI18n(new DatePicker.DatePickerI18n().setDateFormat("dd.MM."));
add(dp);

```

### Steps to reproduce

Create a DatePicker as described above, check the result in the browser. Try to select a value in the calendar popup.

### Environment

Vaadin version(s): 24.5.2
OS: Windows 10 22H2
JDK: OpenJDK 21.0.2

### Browsers

Issue is not browser related

**Comment by gunnarschmid** [comment]
Then you end up having to consider all the logic if date/month/year in combination are valid, see also #6749 

Not supporting years < 10 sounds to me to be a bug rather than a documentation issue.

**Comment by rolfsmeds** [comment]
> Then you end up having to consider all the logic if date/month/year in combination are valid, see also https://github.com/vaadin/flow-components/issues/6749

Yes you do. Regardless, a calendar picker is simply not the appropriate UI component for selecting a month and a date, without a year.

As for y<10 support, let's call it both a bug and a documentation issue.

**Comment by gunnarschmid** [comment]
> Yes you do. Regardless, a calendar picker is simply not the appropriate UI component for selecting a month and a date, without a year.

I totally agree. That makes it so inconvenient that there is no successor to the old DateField component. We have to migrate ca. 50 applications from Vaadin 8 to 24 and most of them use dates in one or the other way.

**Comment by rolfsmeds** [comment]
Indeed, the V8 DateField was in some ways much more flexible.

For your migration, I would recommend implementing your own day+month field, e.g. based on two Selects or ComboBoxes wrapped into a [CustomField](https://vaadin.com/docs/latest/components/custom-field). Checking the validity of the supplied date (e.g. by trying to instantiate a `LocalDate` from it), and showing a validation error when needed, would be fairly easy.

**Comment by web-padawan** [comment]
The formatting part is covered by updating the connector to `createDate()` helper which uses `setFullYear()` internally.

**Comment by web-padawan** [comment]
Related WC issue: https://github.com/vaadin/web-components/issues/1789
